src/Compilers/Core/Portable/Diagnostic/SeverityFilter.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Represents a set of filtered diagnostic severities.
+    /// </summary>
+    internal readonly struct SeverityFilter
```

Comment:
>SeverityFilter [](start = 29, length = 14)

Why a `struct` rather than an `enum` and a `Contains()` extension method?

Reply:
We can also choose a flags enum, just took Jared's suggestion here and went with a simple readonly struct.

Reply:
I'd vote for an `enum` since the type is so simple.

---
In reply to: [414175280](https://github.com/dotnet/roslyn/pull/43546#discussion_r414175280) [](ancestors = 414175280)

Reply:
Sure, let me switch to an enum